src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -165,6 +166,7 @@ export const Configurator = ({
             onConfirm(dataset);
           }}
           fill
+          disabled={timeFields && timeFields.length > 0 && timeFieldName === undefined}
```

Comment:
What will happen if my index does not have any time fields? or if the user has selected the option to not have a time field?

Reply:
Also lets add a unit test for this logic so that we show the fields when we expect the disabled behaviour to show up

Reply:
+1,

the correct flow should be the timeFieldName can be undefined and it should not set it in the dataset.

Honestly I think we should just be consistent with the index pattern creation.

Insert an empty value as the first value and that is the default value and disables the submit.

The `-----` does not set the value of timeFieldName

The `I dont want to use time filter` then enables the submit but ensures the time field name is not set.

<img width="447" alt="Screenshot 2024-09-12 at 1 57 13 PM" src="https://github.com/user-attachments/assets/b75cdfc7-fa6a-44d3-8ad3-b2e32b1dfbfe">